common/utils/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -228,6 +228,19 @@ private[spark] class SparkUnsupportedOperationException private(
   override def getErrorClass: String = errorClass.orNull
 }
 
+private[spark] object SparkUnsupportedOperationException {
+  def apply(): SparkUnsupportedOperationException = {
+    val stackTrace = Thread.currentThread().getStackTrace
+    val messageParameters = if (stackTrace.length >= 4) {
```

Comment:
The stack trace looks like:
<img width="897" alt="Screenshot 2024-01-16 at 19 55 35" src="https://github.com/apache/spark/assets/1580697/9525ff73-bbc6-49eb-88e9-4fe7c5e0c8a4">